arches_lingo/src/arches_lingo/pages/SchemePage.vue

```diff
@@ -1,100 +1,131 @@
 <script setup lang="ts">
-import { ref } from "vue";
+import { computed, markRaw, provide, ref } from "vue";
+
+import { useGettext } from "vue3-gettext";
+import { useRoute } from "vue-router";
+
 import Splitter from "primevue/splitter";
 import SplitterPanel from "primevue/splitterpanel";
-import SchemeLabel from "@/arches_lingo/components/scheme/report/SchemeLabel.vue";
+import SchemeLabel from "@/arches_lingo/components/scheme/SchemeLabel/SchemeLabel.vue";
 import SchemeLicense from "@/arches_lingo/components/scheme/report/SchemeLicense.vue";
 import SchemeNote from "@/arches_lingo/components/scheme/report/SchemeNote.vue";
 import SchemeNamespace from "@/arches_lingo/components/scheme/report/SchemeNamespace.vue";
 import SchemeStandard from "@/arches_lingo/components/scheme/report/SchemeStandard.vue";
 import SchemeEditor from "@/arches_lingo/components/scheme/editor/SchemeEditor.vue";
-import type { SectionTypes } from "@/arches_lingo/types.ts";
-
-const editorVisible = ref(false);
-const sectionVisible = ref(true);
-const editorForm = ref<string>();
-const editorTileId = ref<string>();
-
-const childRefs = ref<Array<SectionTypes>>([]);
-const onMaximize = () => {
-    editorVisible.value = true;
-    sectionVisible.value = false;
-};
-
-const onSide = () => {
-    editorVisible.value = true;
-    editorVisible.value = true;
-    sectionVisible.value = true;
-};
-
-const onClose = () => {
-    editorVisible.value = false;
-    sectionVisible.value = true;
-};
-
-const onOpenEditor = (form: string, tileId: string) => {
-    editorForm.value = form;
-    editorVisible.value = true;
-    sectionVisible.value = true;
+import {
+    CLOSED,
+    MAXIMIZED,
+    MINIMIZED,
+    NEW,
+    VIEW,
+} from "@/arches_lingo/constants.ts";
+
+provide("openEditor", openEditor);
+provide("forceSectionRefresh", forceSectionRefresh);
+
+const route = useRoute();
+const { $gettext } = useGettext();
+
+const editorPanelKey = ref(0);
+const editorTileId = ref();
+const editorState = ref(CLOSED);
+const selectedComponentDatum = ref();
+
+const resourceInstanceId = computed<string | undefined>(() => {
+    if (route.params.id !== NEW) {
+        return route.params.id as string;
+    }
+
+    return undefined;
+});
+
+const componentData = ref([
+    {
+        component: markRaw(SchemeLabel),
```

Comment:
`markRaw` has a big skull and crossbones in the docs. What about just not making `componentData` reactive at all? (I'm not seeing the need for reactivity at this juncture.)

Reply:
ah, the reactivity is there to support `key`-based refreshing. I get the skull && crossbones but this _is_ one of the scenarios the docs described for it.